packages/driver/src/dom/visibility/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,330 @@
+| **[viewport-scenarios](../../../cypress/fixtures/visibility/overflow.html)** | Absolutely positioned element outside of the viewport | ✅ Yes | ❌ No | ❌ No | Elements that are outside of the viewport must be scrolled to before the fast algorithm will consider them visible. This is aligned with scroll-container visibility. |
+| **[z-index-coverage](../../../cypress/fixtures/visibility/positioning.html)** | Covered by higher z-index element |  ✅ Yes | ❌ No | ❌ No | Element covered by another element with higher z-index |
+| **[clip-scenarios](../../../cypress/fixtures/visibility/overflow.html)** | Element clipped by CSS clip property |  ✅ Yes | ❌ No | ❌ No | Element with `clip: rect(0, 0, 0, 0)` or similar clipping |
+| **[transform](../../../cypress/fixtures/visibility/transforms.html)** | Element transformed outside viewport | ✅ Yes | ❌ No | ❌ No | Element with `transform: translateX(-9999px)` or similar |
+| **[contain](../../../cypress/fixtures/visibility/basic-css-properties.html)** | Element with CSS contain:paint property | ✅ Yes | ❌ No | ❌ No | Element positioned outside of a parent that has the `contain: paint` property |
+| **[backdrop-filter](../../../cypress/fixtures/visibility/basic-css-properties.html)** | Element covered by an element with with backdrop-filter opacity(0) | ✅ Yes | ❌ No | ❌ No|  |
+| **[pointer-events-none](../../../cypress/fixtures/visibility/basic-css-properties.html)** | Element with pointer-events: none | ✅ Yes | ❌ No | ❌ No | Element has dimensions and is visible to the user, but cannot receive pointer events. |
+
+## Migration Steps
+
+### Step 1: Enable Fast Visibility
+```javascript
+// cypress.config.js
+module.exports = {
+  e2e: {
+    experimentalFastVisibility: true
+  }
+}
+```
+
+**Note**: Ensure your application under test does not extensively use custom Shadow DOM elements, as the fast visibility algorithm does not yet support Shadow DOM.
+
+### Step 2: Run Your Tests
+Run your existing test suite to identify any failures:
+
+```bash
+npm run test
+```
+
+### Step 3: Analyze Failures
+Look for tests that fail with visibility-related assertions. Common patterns:
+
+```javascript
+// These assertions may behave differently
+.should('be.visible')
+.should('not.be.hidden')
+.is(':visible')
+.is(':hidden')
+```
+
+### Step 4: Disable Fast Visibility for Failing Specs
+For specs that fail due to Shadow DOM or other incompatibilities, disable fast visibility:
+
+```javascript
+// In failing spec files
+describe('My Test Suite', { experimentalFastVisibility: false }, () => {
+  it('should work with legacy visibility', () => {
+    // Your test here
+  })
+})
+```
+
+This allows you to gradually migrate specs while keeping failing ones working.
+
+### Step 5: Fix Tests with Visibility-Related Failures
+For specs that fail due to visibility algorithm differences, update the test expectations to match the correct behavior (see solutions below).
```

Comment:
I might mention: correct behavior = visually look at it and see if it is visible or not!